bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,239 @@
+     */
+    public FloureonThermostatHandler(Thing thing) {
+        super(thing);
+        try {
+            blDevice = new FloureonDevice(host, new Mac(mac));
+            this.floureonDevice = (FloureonDevice) blDevice;
+        } catch (IOException e) {
+            logger.error("Could not find broadlinkthermostat device at Host {} with MAC {} ", host, mac, e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command class: {}", command.getClass());
+        logger.debug("Command: {}", command.toFullString());
+
+        if (command == RefreshType.REFRESH) {
+            refreshData();
+            return;
+        }
+
+        switch (channelUID.getIdWithoutGroup()) {
+            case SETPOINT:
+                handleSetpointCommand(channelUID, command);
+                break;
+            case POWER:
+                handlePowerCommand(channelUID, command);
+                break;
+            case MODE:
+                handleModeCommand(channelUID, command);
+                break;
+            case SENSOR:
+                handleSensorCommand(channelUID, command);
+                break;
+            case REMOTE_LOCK:
+                handleRemoteLockCommand(channelUID, command);
+                break;
+            case TIME:
+                handleSetTimeCommand(channelUID, command);
+                break;
+            default:
+                logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handlePowerCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            try {
+                floureonDevice.setPower(command == OnOffType.ON);
+            } catch (Exception e) {
+                logger.error("Error while setting power of {} to {}", thing.getUID(), command, e);
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleModeCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof StringType) {
+            try {
+                if (MODE_AUTO.equals(command.toFullString())) {
+                    floureonDevice.switchToAuto();
+                } else {
+                    floureonDevice.switchToManual();
+                }
+            } catch (Exception e) {
+                logger.error("Error while setting power of {} to {}", thing.getUID(), command, e);
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleSetpointCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof QuantityType) {
+            try {
+                floureonDevice.setThermostatTemp(((QuantityType) command).doubleValue());
```

Comment:
You should use `QuantityTyp.toUnit()` to convert the temperature to the unit you expect. The user could use °F or °C.

Reply:
Changed to
`floureonDevice.setThermostatTemp(((QuantityType) command).toUnit(SIUnits.CELSIUS).doubleValue());`
Is this the correct way?

Reply:
Yes, but you need to check if `toUnit()` returned null, in case the unit couldn't be converted.

Reply:
Thanks, I added the null check.